CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+cmake_minimum_required(VERSION 3.20)
+project(AllTest)
+enable_testing()
+
+set(EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_SOURCE_DIR}/build)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
+
+include_directories(${GMOCK_ROOT}/include)
+include_directories(${GMOCK_ROOT}/gtest/include)
+include_directories(${GTEST_INCLUDE_DIRS})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src/Header)
+
+find_package(GTest REQUIRED)
+
+include(GoogleTest)
+
+if(UNIX)
+#	add_executable(TestRequestParser 
+#			src/Implement/UnitTestRequestParser.cpp
+#			src/Implement/RequestParser.cpp
+#			src/Header/RequestParser.h
+#			src/Header/OperateData.h
+#			)
+#	target_link_libraries(TestRequestParser pthread GTest::gtest GTest::gtest_main GTest::gmock GTest::gmock_main)
+#	gtest_discover_tests(TestRequestParser)
+#	add_test(RequestParser_test TestRequestParser)
+
+#	add_executable(TestHttpListener 
+#			src/Implement/UnitTestHttpListener.cpp
+#			src/Implement/HttpListener.cpp
+#			src/Header/HttpListener.h
+#			)
+#	target_link_libraries(TestHttpListener pthread GTest::gtest GTest::gtest_main GTest::gmock GTest::gmock_main)
+#	gtest_discover_tests(TestHttpListener)
+#	add_test(HttpListener_test TestHttpListener)
+
+#	add_executable(TestRequestHandler 
+#			src/Implement/UnitTestRequestHandler.cpp
+#			src/Implement/RequestParser.cpp
+#			src/Header/RequestParser.h
+#			src/Implement/RequestHandler.cpp
+#			src/Header/RequestHandler.h
+#			src/Header/OperateData.h
+#			)
+#	target_link_libraries(TestRequestHandler pthread GTest::gtest GTest::gtest_main GTest::gmock GTest::gmock_main)
+#	gtest_discover_tests(TestRequestHandler)
+#	add_test(RequestHandler_test TestRequestHandler)
+endif(UNIX)
```

Comment:
Consider adding else and cause it to fail.

Reply:
Done